### ISSUE TITLE
Crash due to commit c1364d26249431427e663a4b6f948f2fee3549f5

### ISSUE THREAD

**Issue by ampli** [opencog/link-grammar#63]
It happens on a sentence from en/4.0.fixes.batch, but only in batch mode.

Repeat by:
link-parser -batch
They pillory whoever they want to.
^D

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff4da981d in compute_chosen_words (sent=0x601a00007bc0, linkage=**0x0**, opts=0x60180000be00)
    at /usr/local/src/link-grammar-devel/test/link-grammar/linkage.c:93

The **linkage** argument is fine if not in batch mode.

**Comment by linas** [comment]
fixed with pull request #64

**Comment by linas** [comment]
As a side effect, it was impossible to print bad linkages when !bad was set.

**Comment by linas** [comment]
I guess this means that there will be a version 5.2.5 in a few days or a week or so. (thus delaying the merge of the wordgraph code some more). Any fixes you want to have for that?

**Comment by ampli** [comment]
I finished the simplification and correction of the dictopen stuff.
Even though it is not related to parsing at all, I tried the batch tests "just in case" (and found this crash).
I would like to test it some more (opening dicts in different strange paths).
Because the fix involves handling paths, I also would like to test it on native Windows, if possible (I need to set up a compilation environment, but I don't have an experience in that, so I'm not sure I will be able to do it in a timely manner).
I also have some more other small fixes.
I will try to send them in 2-3 days.
